src/ray/common/client_connection.cc

```diff
@@ -602,4 +607,32 @@ std::string ServerConnection::DebugString() const {
   return result.str();
 }
 
+std::vector<bool> CheckForClientDisconnects(
+    const std::vector<std::shared_ptr<ClientConnection>> &conns) {
```

Comment:
This is a standalone utility instead of a method on `ClientConnection` so we can check them in a batch w/ a single syscall